packages/frontend-api/src/Model/Complaint/ComplaintApiFacade.php

```diff
@@ -58,4 +85,88 @@ public function create(ComplaintData $complaintData): Complaint
 
         return $complaint;
     }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @return \Shopsys\FrameworkBundle\Model\Complaint\Complaint
+     */
+    public function createFromComplaintInputArgument(Argument $argument): Complaint
+    {
+        $input = $argument['input'];
+
+        $order = $this->orderApiFacade->getByUuid($input['orderUuid']);
+        $customerUser = $this->currentCustomerUser->findCurrentCustomerUser();
+        $orderCustomerUser = $order->getCustomerUser();
+
+        if (!($orderCustomerUser === $customerUser || (
+            $this->security->isGranted(CustomerUserRole::ROLE_API_ALL) &&
+                $customerUser->getCustomer() === $order->getCustomer()
+        ))
+        ) {
+            throw new InvalidAccessUserError('You are not allowed to create complaint for this order');
+        }
+
```

Comment:
Extracting the parts of the condition into variables would improve the readability, IMHO. Also, I believe we need to check for the nullability of `$customerUser` as only logged users can create complaints, right? (Otherwise, any anonymous user would be able to create a complaint for anonymous order). What about this?
```
$isSameCustomerUser = $customerUser !== null && $orderCustomerUser === $customerUser;
$isCustomerAccountOwner = $this->security->isGranted(CustomerUserRole::ROLE_API_ALL) && $customerUser->getCustomer() === $order->getCustomer();

if (!$isSameCustomerUser && !$isCustomerAccountOwner) {
    throw new InvalidAccessUserError('You are not allowed to create complaint for this order');
}
```

Reply:
fixed [fixup! fixup! complaints b2b customers support](https://github.com/shopsys/shopsys/pull/3295/commits/cf1dc4153034074866c7924fd034d3dbf959492b)